### ISSUE TITLE
A11y_.NET Core_WPF_ExpenseItIntro_ExpenseReportfor_Keyboard: The table column headers are not accessible by keyboard.

### ISSUE THREAD

**Issue by v-inanda** [dotnet/wpf#7850]
### Description

“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

### GitHubTags: 
#A11yMAS;#A11yTCS;#BM_.NET Core[WPF]_Win32_May23;#.NET Core;#A11ySev2;#WCAG2.1.1;#DesktopApp;#Win11;#Keyboard;#Benchmark;

### Environment Details:
Application Name: .NET Core
.NET Version: .NET-SDK-8.0.100-preview.4.23260.5
 WPF
Operating System: Windows 11 Enterprise 21H2

### Reproduction Steps:
1. Launch VS 2022 Preview
2. TAB to Getting started and right click on the Expense it intro and click on build
3. Then right click on Expense it Intro then Click on Debug-->Start New Instance.
4. Navigate to Names list and select any name and click on View.
5. 'Expense Report for' screen will open
6. Tab till table and observe the issue.

### Expected behavior

The table column headers "ExpenseType" and "Amount" should be accessible by keyboard. So that the sorting functionality should be accessible by using keyboard.

### Actual behavior

The table column headers "ExpenseType" and "Amount" are not accessible by keyboard. While using mouse and clicking on this column headers sorting functionality is performed but we are not able to use sorting functionality by using keyboard
![A11y_ NET Core_WPF_ExpenseItIntro_ExpenseReportfor_Keyboard](https://github.com/dotnet/wpf/assets/91525824/f8b7397b-4a64-427b-927d-1169f373a9c2)

### Impact

Keyboard users will not be able to sort the data by using keyboard if we are not able to use sorting functionality by using keyboard.

**Comment by singhashish-wpf** [comment]
@v-inanda Please have a look at this bug: https://github.com/dotnet/wpf/issues/6737
It is the same bug which has all the discussions.

**Comment by miloush** [comment]
@singhashish-wpf Well, looks like the shortcut didn't really help... I suggest we revert the shortcut and make the datagrid header focusable using Tab as expected. This would enable way more accessibilty scenarios (context menus on the headers, selecting columns etc.).

**Comment by merriemcgaw** [comment]
@singhashish-wpf - I know you added the ability to do the sorting by column, but did you update the sample app to use it? That would resolve this bug. On the other hand, the longer-term solution may well be to create selectable headers.

@v-inanda there was a feature introduced as a result of the duplicate bug in an earlier released that did cause a lot of trouble. I don't know the state of sorting columns in .NET 8, but @singhashish-wpf can tell you what keyboard command may work for this sample app.

**Comment by singhashish-wpf** [comment]
@merriemcgaw The sample app is updated via the PR https://github.com/microsoft/WPF-Samples/pull/498
We can fetch the latest changes from the wpf-samples repo and see the functionality. Let me know if you have any further questions.

**Comment by v-inanda** [comment]
@harshit7962 Verified the issue and it is fixed. Hence, closing the bug.

![Presentation1](https://github.com/dotnet/wpf/assets/91525824/bfc30723-8305-496c-9dc3-6a1865e4f3eb)
